server2/Arcadia.Assistant/Arcadia.Assistant.Sharepoint/SharepointSynchronizator.cs

```diff
@@ -0,0 +1,171 @@
+﻿namespace Arcadia.Assistant.Sharepoint
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    using CalendarEvent;
+
+    using Employees.Contracts;
+
+    using ExternalStorages.Abstractions;
+
+    using Microsoft.Extensions.Logging;
+
+    using SickLeaves.Contracts;
+
+    using Vacations.Contracts;
+
+    using WorkHoursCredit.Contracts;
+
+    public sealed class SharepointSynchronizator
+    {
+        #region ctor
+
+        public SharepointSynchronizator(ISickLeaves sickLeaves, IVacations vacations, IWorkHoursCredit workouts, ISharepointDepartmentsCalendarsSettings departmentsCalendarsSettings, ILogger? logger)
+        {
+            this.sickLeaves = sickLeaves;
+            this.vacations = vacations;
+            this.workouts = workouts;
+            this.departmentsCalendarsSettings = departmentsCalendarsSettings;
+            this.logger = logger;
+        }
+
+        #endregion
+
+        #region publci interface
+
+        public async Task Synchronize(IEmployees employees, IEnumerable<string> departments, IExternalStorage storage, CancellationToken cancellationToken)
+        {
+            var vacationsSync = new EmployeeVacationsSynchronization(storage, this.logger);
+            var workoutsSync = new EmployeeWorkoutsSynchronization(storage, this.logger);
+            var sickLeavesSync = new EmployeeSickLeavesSynchronization(storage, this.logger);
+
+            var sharepointCalendars = departments.Distinct().ToDictionary(x => x,
```

Comment:
Naming is confusing. It should better be `sharepointItemsByCalendar`, because you already have method `GetSharepointCalendarsByDepartment` and my first thought was they are related somehow.

Reply:
Done